csrc/scheduler/normalization_utils.cpp

```diff
@@ -853,29 +855,36 @@ bool isProjectBufferToInputs(
     const scheduler_utils::PersistentBufferInfo& persistent_buffer_info,
     const scheduler_utils::PersistentBufferSizeReturn&
         persistent_buffer_size_info,
-    const bool is_inner_reduction) {
+    const ScheduleHeuristic sh,
+    const bool is_inner_outer_with_outer_bcast) {
   // don't project if there are view ops and no buffer can be projected
   bool can_project = ir_utils::getViewOps(fusion).empty() &&
       persistent_buffer_size_info.projected_persistent_buffer_size > 0;
   if (!can_project) {
     return false;
   }
 
-  // don't project if can't reduce buffer size
-  if (persistent_buffer_size_info.projected_persistent_buffer_size >=
-      persistent_buffer_size_info.persistent_buffer_size) {
+  // don't project if can't reduce buffer size unless it's innerOuter with
+  // outer broadcast where project to inputs reduces gmem access.
+  if (!is_inner_outer_with_outer_bcast &&
```

Comment:
nitpick: rename `is_inner_outer_with_outer_bcast` to `check_projected_buffer_size` or something like that, and put the comment `unless it's innerOuter with outer broadcast where project to inputs reduces gmem access.` to the place where it's called.

The util here is shared across normalization scheduler and I *think* that's a more readable argument name.

Reply:
Changed, thanks!